### PR TITLE
Move imports around to make raider-base cleaner

### PR DESCRIPTION
**Pull request by jlmaurer**
Refactor raider to make the raider-base lighter

## Description

- By cleaning up import statements and moving isolated imports to their calling functions, this PR provides for a more flexible RAiDER that **may** not need as many 3rd party package installs. 
- TBD: 
   - clean up conda installs 
   - add pre-processing functionality for nisar
- I'll update this PR with these TBDs but wanted to post now to solicit feedback on the general idea

## Motivation and Context

- Addresses #533

## How Has This Been Tested?

- Not tested yet. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have added an explanation of what your changes do and why you'd like us to include them.
- [ ] I have written new tests for your core changes, as applicable.
- [ ] I have successfully ran tests with your changes locally.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by bbuzz31** [comment]
This needs to be rebased I believe

**Comment by cmarshak** [comment]
David asked me to review. I can only say delaying import errors looks good.

```
try:
     import package
except ImportError
    package = None

def func():
   if not package:
      raise ImportError('Package not imported')
```

File: test/test_synthetic.py [hunk 1]
```diff
@@ -193,7 +191,7 @@ def test_dl_real(region, mod='ERA5'):
     proc = subprocess.run(cmd.split(), stdout=subprocess.PIPE, universal_newlines=True)
     assert proc.returncode == 0, 'RAiDER did not complete successfully'
 
-
+@pytest.mark.long()
```
Comment (bbuzz31) [inline test/test_synthetic.py:194]: @jlmaurer can we remove these long? The weather models are staged so they take like 2 min each max

**Review by bbuzz31 — CHANGES_REQUESTED** [review]

File: test/test_synthetic.py (hunk 1 above)
Comment (jlmaurer) [reply test/test_synthetic.py:194]: @bbuzz31 these take forever on my system...are they staged here?

**Review by bbuzz31 — APPROVED** [review]

**Pull request merged**